store/src/main/java/com/nytimes/android/external/store/base/impl/RealInternalStore.java

```diff
@@ -265,6 +266,14 @@ void updateMemory(@Nonnull final BarCode barCode, final Parsed data) {
     @Override
     public void clearMemory() {
         memCache.invalidateAll();
+        inFlightRequests.invalidateAll();
```

Comment:
Shouldn't you clear the inflight requests first in case there is one that successes and writes in memory after you have cleared it and before you invalidate the inFlight ones?